### DISCUSSION TITLE
Upgrading from 2 to 3

### DISCUSSION THREAD

**Discussion by lzoubek** [ionic-team/capacitor#5303] [Q&A] [answered]
Hello,I have a plan to upgrade my app from cap 2 to cap 3. I went through the guide and the process is pretty obvious.

I am using cordova-plugin-ionic  to keep the app ever green.

Upgrade to 3 can look like:
Migrate code base from 2 to 3
Publish new code so  cordova-plugin-ionic can pick it up
Publish new app to app stores

Q1: What happens when user updates application from app store, but his HTML/JS is still running code based on Cap 2 ? I'd need to app to at least start so cordova-plugin-ionic can update to new code

Q2: Can Cap 3 based JS code run on top of Cap 2?

**Comment by jcesarmobile** [comment, marked as answer]
The first launch after you bump the app version, the app launches with the compiled assets, not with the downloaded assets.

Then depending on how you configured deploy, it might be possible that the app downloads an older version or launches from from a downloaded version on subsequent loads, in example if you didn't deploy the same version to Appflow so latest version is still the 2.x version.
There could also be the problem that users are still on version 2 but you serve them version 3 assets and that can cause problems.

I think best thing in this cases is to create a new channel for v3 (called destinations on the dashboard), so v3 apps only check updates from that channel and v2 apps keep checking the old channel so they don't get v3 updates.
